### DISCUSSION TITLE
Is it possible to call listen on Next.JS Server without a custom server?

### DISCUSSION THREAD

**Discussion by rubenferreira97** [vercel/next.js#40258] [Help]
I am deploying a server side Next.JS app on CPanel, which needs a entry/startup file.

Currently I am doing like this:
```
import next from 'next';
import { createServer } from 'http';

const app = next();

async function start() {
  await app.prepare();
  const handle = app.getRequestHandler();
  createServer(handle).listen(3000);
}

start();
```
Is it possible to listen the Next.js server instance without a custom server? I assume Next.js server is optimized for a Next.JS app, unlike `http.createServer`.

**Comment by schizek** [comment]
Were you ever able to figure this out?   I'm looking for the answer to this as well.

Here is a similar question from over a year ago (Nov 2021) which is still unanswered... https://github.com/vercel/next.js/discussions/31372 ...so perhaps I shouldn't hold my breath. 😬 As of right now the only way I have Next.js running in cPanel is by using Application Manager which by default points to an app.js file in my docroot that I have set with the same contents as the server.js file suggested in the [Custom Server](https://nextjs.org/docs/advanced-features/custom-server) documentation.
